File: 360MedLink-Back/src/main/java/org/sid/entities/Conge.java
package org.sid.entities;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.validation.constraints.NotEmpty;
import java.time.LocalDate;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
@ToString
public class Conge {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    @NotEmpty(message = "managerName is required")
    private String    managerName;
    @NotEmpty(message = "firstName is required")
    private String    firstName;
    @NotEmpty(message = "SecondeName is required")
    private String    secondeName;
    @NotEmpty(message = "titre is required")
    private String    titre;
    private Integer   idUser;
    private Integer   managerId;
    private LocalDate debutTime;
    private Integer   breJours;
    private String    status;
    private Integer   rhApprouvment;

}

File: 360MedLink-Front/src/app/models/conge.ts
export class Conge{
  public  id :number;
  public  managerName: string;
  public  idUser:      number;
  public  manager_id:  number;
  public  debutTime:   string;
  public  breJours :   number;
  public  firstName:   string;
  public  secondeName: string;
  public  titre       :string;
  public  status      :string;
  public  rhApprouvment:string;

  constructor(id?:number,managerName?: string, idUser?: number, manager_id?: number, debutTime?: string, breJours?: number,
              firstName?: string, secondeName?: string, titre?: string) {
    this.managerName = managerName="";
    this.idUser = idUser=null;
    this.manager_id = manager_id=null;
    this.debutTime = debutTime="";
    this.breJours = breJours=0;
    this.firstName = firstName="";
    this.secondeName = secondeName="";
    this.titre = titre="";
  }
}
